app/controlplane/plugins/core/webhook/v1/webhook.go

```diff
@@ -0,0 +1,327 @@
+			if err != nil {
+				i.Logger.Errorw("failed to marshal SBOM", "error", err, "type", material.Type)
+				return fmt.Errorf("marshalling SBOM: %w", err)
+			}
+			if err := i.sendWebhook(ctx, webhookURL, material.Type, json.RawMessage(materialJSON), req.ChainloopMetadata); err != nil {
+				i.Logger.Errorw("failed to send SBOM webhook", "error", err, "type", material.Type)
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// sendWebhook sends a webhook with the specified kind and payload
+func (i *Integration) sendWebhook(ctx context.Context, url, kind string, payload json.RawMessage, metadata *sdk.ChainloopMetadata) error {
+	payloadBytes, err := json.Marshal(webhookPayload{
+		Metadata: metadata,
+		Data:     payload,
+		Kind:     kind,
+	})
+	if err != nil {
+		i.Logger.Errorw("failed to marshal webhook payload", "error", err, "kind", kind)
+		return fmt.Errorf("marshalling webhook payload: %w", err)
+	}
+
+	headers := map[string]string{
+		"Content-Type": "application/json",
+	}
+
+	resp, err := i.doHTTPRequest(ctx, http.MethodPost, url, payloadBytes, headers)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			i.Logger.Warnw("failed to close response body", "error", err)
+		}
+	}()
+
+	// Read response body for more detailed error messages
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		i.Logger.Warnw("failed to read response body", "error", err)
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		i.Logger.Errorw("webhook responded with non-success status code", "status", resp.StatusCode, "body", string(body))
+		return fmt.Errorf("webhook responded with status code %d: %s", resp.StatusCode, string(body))
+	}
+
+	return nil
+}
+
+// doHTTPRequest is a helper method to perform HTTP requests
+func (i *Integration) doHTTPRequest(ctx context.Context, method, url string, body []byte, headers map[string]string) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, method, url, bytes.NewBuffer(body))
+	if err != nil {
+		i.Logger.Errorw("failed to create HTTP request", "error", err, "url", url)
+		return nil, fmt.Errorf("creating HTTP request: %w", err)
+	}
+
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+
+	resp, err := i.client.Do(req)
+	if err != nil {
+		i.Logger.Errorw("failed to send HTTP request", "error", err, "url", url)
+		return nil, fmt.Errorf("sending HTTP request: %w", err)
+	}
+
+	return resp, nil
+}
+
+// testWebhookURL sends a POST request to ensure the webhook URL is reachable
+func (i *Integration) testWebhookURL(ctx context.Context, webhookURL string) error {
+	// fake payload with Metadata, Data and Kind
+	testPayload := []byte(`{"Metadata":{"Workflow":"CHAINLOOPTEST","WorkflowRun":"CHAINLOOPTEST"},"Data":{},"Kind":"CHAINLOOPTEST"}`)
+	headers := map[string]string{
+		"Content-Type": "application/json",
+	}
+
+	resp, err := i.doHTTPRequest(ctx, http.MethodPost, webhookURL, testPayload, headers)
```

Comment:
is there any way you could use sendWebhook instead mocking some of the data? that way you have a single way and ensure the same payload